src/caja-query-editor.c

```diff
@@ -999,6 +1025,231 @@ type_add_rows_from_query (CajaQueryEditor    *editor,
+}
+
+static void modtime_row_free_data(CajaQueryEditorRow *row)
+{
+}
+
+static void modtime_add_rows_from_query(CajaQueryEditor *editor, CajaQuery *query)
+{
+}
+
+
+static GtkWidget *size_row_create_widgets(CajaQueryEditorRow *row)
+{
+    GtkWidget *hbox = NULL;
+    GtkWidget *combo = NULL;
+    GtkWidget *size_combo = NULL;
+    GtkCellRenderer *cell = NULL;
+    GtkListStore *store = NULL;
+    GtkListStore *size_store = NULL;
+    GtkTreeIter iter;
+
+    hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 7);
+
+    store = gtk_list_store_new(2, G_TYPE_BOOLEAN, G_TYPE_STRING);
+    combo = gtk_combo_box_new_with_model(GTK_TREE_MODEL(store));
+    g_object_unref(store);
+
+    cell = gtk_cell_renderer_text_new();
+    gtk_cell_layout_pack_start(GTK_CELL_LAYOUT(combo), cell, TRUE);
+    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(combo), cell, "text", 1,
+                                   NULL);
+
+    gtk_list_store_append(store, &iter);
+    gtk_list_store_set(store, &iter, 0, FALSE, 1, _("Less than or equal to"), -1);
+    gtk_list_store_append(store, &iter);
+    gtk_list_store_set(store, &iter, 0, TRUE, 1, _("Greater or equal to"), -1);
+
+    gtk_combo_box_set_active(GTK_COMBO_BOX(combo), 0);
+
+    size_store = gtk_list_store_new(2, G_TYPE_INT64, G_TYPE_STRING);
+    size_combo = gtk_combo_box_new_with_model(GTK_TREE_MODEL(size_store));
+    g_object_unref(size_store);
+
+    gtk_cell_layout_pack_start(GTK_CELL_LAYOUT(size_combo), cell, TRUE);
+    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(size_combo), cell, "text",
+                                   1, NULL);
+
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 10240, 1, _("10 KB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 102400, 1, _("100 KB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 512000, 1, _("500 KB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 1048576, 1, _("1 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 5242880, 1, _("5 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 10485760, 1, _("10 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 104857600, 1, _("100 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 524288000, 1, _("500 MB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 1073741824, 1, _("1 GB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 2147483648, 1, _("2 GB"), -1);
+    gtk_list_store_append(size_store, &iter);
+    gtk_list_store_set(size_store, &iter, 0, 4294967296, 1, _("4 GB"), -1);
```

Comment:
Agree with the comments above. These units (and their value) should reflect the users preference